### PR TITLE
Fix Java PreCommit Action correct branch to run

### PR DESCRIPTION
**Pull request by Abacn**
Found in https://github.com/apache/beam/pull/28434#issuecomment-1718278586

Also addressing https://github.com/apache/beam/pull/28075#issuecomment-1714652977

* Also upload human readable test reports on failure

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @andreydevyatkin @damccorm

File: .github/workflows/beam_PreCommit_Java.yml [hunk 1]
```diff
@@ -172,10 +173,11 @@ jobs:
             -PdisableSpotlessCheck=true \
             -PdisableCheckStyle=true \
       - name: Upload test report
+        if: failure()
```
Comment (damccorm) [inline .github/workflows/beam_PreCommit_Java.yml:176]: Should this be `if: always()`?

**Comment by Abacn** [comment]
using a script checked that this the only one missed `./.github/actions/setup-action` (by searching `uses: actions/checkout` and no `.github/actions/setup-action`)

File: .github/workflows/beam_PreCommit_Java.yml (hunk 1 above)
Comment (Abacn) [reply .github/workflows/beam_PreCommit_Java.yml:176]: There will be a third party workflow generating junit reports: #28075 to be added to all tests. This is available as a complement if one ones to see raw reports so failure() should suffice. I did not investigate how big the artifact could be so I would suggest roll out to failure() first

Comment (damccorm) [reply .github/workflows/beam_PreCommit_Java.yml:176]: Ok, SGTM

**Review by damccorm — APPROVED** [review]

**Pull request merged**